pyiron/atomistics/structure/atoms.py

```diff
@@ -2159,7 +2159,10 @@ def __delitem__(self, key):
         if isinstance(key, (int, np.integer)):
             key = [key]
         key = np.array(key).flatten()
-        new_length = len(self) - len(key)
+        if isinstance(key[0], (bool, np.bool_)):
```

Comment:
I'm fine with this implementation, but if you want to reduce the number of lines, you could also do

```
new_length = len(self) - len(np.arange(len(key))[np.asarray(key)])
```

Reply:
This is much nices (although a small change is required `new_length = len(self) - len(np.arange(len(self))[np.asarray(key)])`)